experimental/PropertyDDS/packages/property-properties/src/properties/arrayProperty.js

```diff
@@ -403,21 +403,27 @@ export class ArrayProperty extends AbstractStaticCollectionProperty {
                 AbstractStaticCollectionProperty.prototype.setValues.call(this, in_values);
             }
         } else {
-            if (_.isArray(in_values) && in_values.length < this._dataArrayGetLength()) {
-                this.removeRange(in_values.length, this._dataArrayGetLength() - in_values.length);
+          if (_.isArray(in_values)) {
+            if (in_values.length < this._dataArrayGetLength()) {
+              this.removeRange(in_values.length, this._dataArrayGetLength() - in_values.length);
             }
-
+            this.setRange(0, in_values.slice(0, this._dataArrayGetLength()));
+            if (in_values.length > this._dataArrayGetLength()) {
+              this.insertRange(this._dataArrayGetLength(), in_values.slice(this._dataArrayGetLength()));
+            }
+          } else {
             var that = this;
             var maxIndex = this._dataArrayGetLength() - 1;
             _.each(in_values, function(value, index) {
```

Comment:
The non-array case is used to handle sparse updates. This is an example from the tests:

```
MyArrayProp.setValues({
  0: 12,
  2: 9,
});
```

Converting those to an array, would cause the indices to be lost.

Reply:
Makes sense (and that's a neat feature!)